core/src/main/java/org/elasticsearch/index/codec/PerFieldMappingPostingFormatCodec.java

```diff
@@ -56,7 +57,11 @@ public PerFieldMappingPostingFormatCodec(Lucene50StoredFieldsFormat.Mode compres
     public PostingsFormat getPostingsFormatForField(String field) {
         final MappedFieldType indexName = mapperService.indexName(field);
         if (indexName == null) {
-            logger.warn("no index mapper found for field: [{}] returning default postings format", field);
+            // the query terms are an percolator implementation detail and are added as Lucene indexed fields which are hidden
```

Comment:
@bleskes somehow in GH this comment is misplaced :) but I'm not sure if I follow... Although we don't do realtime percolation any more, we do need to add the extracted query terms as extra Lucene fields to the percolator document. How would we do that if don't intercept the document just before the actual indexing?